### PR TITLE
[BugFix: Issue #280] Add support for rendering provider documentation including Terraform 0.13 required_providers section

### LINKED ISSUE #280: Terraform docs generator does not include info about configuring terraform required_providers section
## Describe the bug

The OpenAPI Terraform provider supports Terraform 0.12 and also greater version >= 0.13 <0.15 at the moment. Terraform 0.13 and greater now require that each Terraform module must declare which providers it requires, so that Terraform can install and use them. Provider requirements are declared in a required_providers block. However the documentation rendered is missing this information.

Note: The service provider is responsible for installing the in-house provider in the corresponding location that Terraform expects plugins to be located at. Terraform 0.12 and Terraform >=0.13 have different requirements in terms of where the custom plugins should be installed. The following script may be used to automate the plugin provisioning task which depending on what version of Terraform the user is using the plugin will be installed accordingly in the expected location for Terraform: https://github.com/dikhan/terraform-provider-openapi/blob/master/scripts/install.sh

## To Reproduce
A clear and concise description on how does it work now and what's the current behaviour.

1. Running the `/Users/dkhanram/dev/SourceCode/src/github.com/dikhan/terraform-provider-openapi/pkg/terraformdocsgenerator/main.go` application will generate a file `example_provider_documentation_output.html` containing the Provider Installation but missing the Terraform 0.13 required configuration.

````
</ul><h2 id="provider_installation">Provider Installation</h2>
<p>
  In order to provision 'openapi' Terraform resources, you need to first install the 'openapi'
  Terraform plugin by running&nbsp;the following command (you must be running Terraform &gt;= 0.12):
</p>
<pre>$ export PROVIDER_NAME=openapi && curl -fsSL https://raw.githubusercontent.com/dikhan/terraform-provider-openapi/master/scripts/install.sh | bash -s -- --provider-name $PROVIDER_NAME<br>[INFO] Downloading https://github.com/dikhan/terraform-provider-openapi/releases/download/v0.29.4/terraform-provider-openapi_0.29.4_darwin_amd64.tar.gz in temporally folder /var/folders/n_/1lrwb99s7f50xmn9jpmfnddh0000gp/T/tmp.Xv1AkIZh...<br>[INFO] Extracting terraform-provider-openapi from terraform-provider-openapi_0.29.4_darwin_amd64.tar.gz...<br>[INFO] Cleaning up tmp dir created for installation purposes: /var/folders/n_/1lrwb99s7f50xmn9jpmfnddh0000gp/T/tmp.Xv1AkIZh<br>[INFO] Terraform provider 'terraform-provider-openapi' successfully installed at: '~/.terraform.d/plugins'!</pre>
<p>
  <span>You can then start running the Terraform provider:</span>
</p>
<pre dir="ltr">$ export OTF_VAR_openapi_PLUGIN_CONFIGURATION_FILE="https://api.service.com/openapi.yaml"<br>
➜ ~ terraform init &amp;&amp; terraform plan
</pre>
````

## Expected behaviour

Example:
```gherkin
Scenario: terraformdocsgenerator renders provider configuration including terraform required_providers (mandatory for Terraform >= 0.13) documentation
Given an OpenAPI document that is OpenAPI Terraform compliant
When I execute the terraform docs generator
Then I should see that in the plugin installation section there is a snippet of code already generated with how users should configure their terraform configuration to be able to use the plugin with Terraform >= 0.13. Example below
```

````
</ul><h2 id="provider_installation">Provider Installation</h2>
<p>
  In order to provision 'openapi' Terraform resources, you need to first install the 'openapi'
  Terraform plugin by running&nbsp;the following command (you must be running Terraform &gt;= 0.12):
</p>
<pre>$ export PROVIDER_NAME=openapi && curl -fsSL https://raw.githubusercontent.com/dikhan/terraform-provider-openapi/master/scripts/install.sh | bash -s -- --provider-name $PROVIDER_NAME<br>[INFO] Downloading https://github.com/dikhan/terraform-provider-openapi/v2/releases/download/v0.29.4/terraform-provider-openapi_0.29.4_darwin_amd64.tar.gz in temporally folder /var/folders/n_/1lrwb99s7f50xmn9jpmfnddh0000gp/T/tmp.Xv1AkIZh...<br>[INFO] Extracting terraform-provider-openapi from terraform-provider-openapi_0.29.4_darwin_amd64.tar.gz...<br>[INFO] Cleaning up tmp dir created for installation purposes: /var/folders/n_/1lrwb99s7f50xmn9jpmfnddh0000gp/T/tmp.Xv1AkIZh<br>[INFO] Terraform provider 'terraform-provider-openapi' successfully installed at: '~/.terraform.d/plugins'!</pre>
<p>
  <span>You can then start running the Terraform provider:</span>
</p>
<pre dir="ltr">$ export OTF_VAR_openapi_PLUGIN_CONFIGURATION_FILE="https://api.service.com/openapi.yaml"<br>
➜ ~ terraform init &amp;&amp; terraform plan
</pre>
<p>
<b>Note:</b> As of Terraform &gt;= 0.13 each Terraform module must declare which providers it requires, so that Terraform can install and use them. If you are using Terraform &gt;= 0.13, copy into your .tf file the 
following snippet already populated with the provider configuration: 
<pre dir="ltr">
terraform {
  required_providers {
    openapi = {
      source  = "terraform.example.com/examplecorp/openapi"
      version = ">= 2.0.1" 
    }
  }
}
</pre>
</p>
````

### Checklist (for admin only)

**Don't forget to go through the checklist to make sure the issue is created properly**:

- [x] I have added a corresponding label (bug) to the issue (right side menu)
- [x] I have added this issue to the 'API Terraform Provider' GitHub project (right side menu)

### PR DESCRIPTION
**Pull request by dikhan**
## What problem does this Pull Request solve?

Please link to the issue number here (issue will be closed when PR is merged): Closes #280 

## Type of change

What type of change does your code introduce to the provider? Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [ ] New feature (change that adds new functionality)
- [x] Bug-fix (change that fixes current functionality)
- [ ] Tech debt (enhances the current functionality)
- [ ] New release (pumps the version)

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have made sure code compiles correctly and all tests are passing by running `make test-all`
- [x] I have added/updated necessary documentation (if appropriate)
- [x] I have added the following info to the title of the PR (pick the appropriate option for the type of change). This is important because the release notes will include this information.
  - [ ] Feature Request: PRs related to feature requests should have in the title `[FeatureRequest: Issue #X] <PR Title>`
  - [x] Bug Fixes: PRs related to bug fixes should have in the title `[BugFix: Issue #X] <PR Title>`
  - [ ] Tech Debt: PRs related to technical debt should have in the title `[TechDebt: Issue #X] <PR Title>` 
  - [ ] New Release: PRs related to a new release should have in the title `[NewRelease] vX.Y.Z`

## Checklist for Admins
- [x] Label is populated
- [x] PR is assigned to the corresponding project
- [x] PR has at least 1 reviewer and 1 assignee

### CODE REVIEW

File: pkg/terraformdocsgenerator/openapiterraformdocsgenerator/openapi_terraform_provider_doc_generator.go [hunk 1]
```diff
@@ -1,33 +1,61 @@
...
+	}, nil
 }
 
 // GenerateDocumentation creates a TerraformProviderDocumentation object populated based on the OpenAPIDocURL documentation
 func (t TerraformProviderDocGenerator) GenerateDocumentation() (TerraformProviderDocumentation, error) {
+	if t.ProviderName == "" {
+		return TerraformProviderDocumentation{}, errors.New("provider name not provided")
+	}
+	if t.Hostname == "" {
+		return TerraformProviderDocumentation{}, errors.New("hostname name not provided, this is required to be able to render the provider installation section containing the required_providers block with the source address configuration in the form of [<HOSTNAME>/]<NAMESPACE>/<TYPE>")
```
Comment (jzafran) [inline pkg/terraformdocsgenerator/openapiterraformdocsgenerator/openapi_terraform_provider_doc_generator.go:51]: Is the `name` here after `hostname` a typo? 

```suggestion
		return TerraformProviderDocumentation{}, errors.New("hostname not provided, this is required to be able to render the provider installation section containing the required_providers block with the source address configuration in the form of [<HOSTNAME>/]<NAMESPACE>/<TYPE>")
```

Comment (dikhan) [reply pkg/terraformdocsgenerator/openapiterraformdocsgenerator/openapi_terraform_provider_doc_generator.go:51]: Yep, pushing a commit with a fix shortly

**Review by jzafran — APPROVED** [review]
LGTM

**Comment by dikhan** [comment]
Running into travis sync issues with PR status reporting, for the ref the build passed and linking the build for the record: https://travis-ci.org/github/dikhan/terraform-provider-openapi/builds/763296939

**Pull request merged**